packages/start-slicemachine/src/lib/listen.ts

```diff
@@ -0,0 +1,56 @@
+import type { Server } from "node:http";
+import type { AddressInfo } from "node:net";
+
+type ListeningServer = {
+	/**
+	 * Metadata that can be used to access the server.
+	 */
+	address: AddressInfo;
+
+	/**
+	 * Closes the server.
+	 */
+	close: () => Promise<void>;
+};
+
+type ListenConfig = {
+	/**
+	 * The port on which to listen. If no port is provided, a random available
+	 * port will be used.
+	 */
+	port?: number;
+};
+
+/**
+ * Listens to a standard `node:http` server.
+ *
+ * @param server - The server to listen to.
+ * @param config - Configuration for the listener.
+ *
+ * @returns Metadata and controls for the server.
+ */
+export const listen = async (
+	server: Server,
+	config: ListenConfig = {},
+): Promise<ListeningServer> => {
+	await new Promise<void>((resolve) => {
+		server.once("listening", () => {
+			resolve();
+		});
+
+		server.listen({ port: config.port });
+	});
+
+	return {
+		address: server.address() as AddressInfo,
```

Comment:
what type is `server.address()` ?

Reply:
It's `string | AddressInfo | null`. We know the server is running from the code above this line, so we can omit `null`. I've never seen `server.address()` return `string` before, so I think it's safe to assert the value is `AddressInfo`.

If you know what kind of value `server.address()` returns when it's a string, please share. If we know what it looks like, we can support that value.